Scripts/Runtime/Data/IDProviderAtomicBurstable.cs

```diff
@@ -0,0 +1,76 @@
+using Anvil.CSharp.Data;
+using System.Threading;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+using UnityEngine;
+
+namespace Anvil.Unity.DOTS.Data
+{
+    /// <summary>
+    /// An <see cref="IDProvider"/> that is compatible with Burst and will get the next ID atomically so it is
+    /// safe to use in threaded context.
+    /// </summary>
+    [BurstCompatible]
+    public readonly unsafe struct IDProviderAtomicBurstable
```

Comment:
Based on the other burstable type variants we have this makes the name consistent.

```suggestion
    public readonly unsafe struct BurstableAtomicIDProvider
```

Reply:
Done